src/gui/addnewtorrentdialog.cpp

```diff
@@ -284,12 +284,21 @@ bool AddNewTorrentDialog::loadTorrentImpl()
         BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(infoHash);
         if (torrent) {
             if (torrent->isPrivate() || m_torrentInfo.isPrivate()) {
-                RaisedMessageBox::warning(this, tr("Torrent is already present"), tr("Torrent '%1' is already in the transfer list. Trackers haven't been merged because it is a private torrent.").arg(torrent->name()), QMessageBox::Ok);
+                RaisedMessageBox::warning(this, tr("Torrent is already present"), tr("Torrent '%1' is already in the transfer list.").arg(torrent->name()) + " " + tr("Trackers haven't been merged because it is a private torrent."), QMessageBox::Ok);
             }
             else {
-                torrent->addTrackers(m_torrentInfo.trackers());
-                torrent->addUrlSeeds(m_torrentInfo.urlSeeds());
-                RaisedMessageBox::information(this, tr("Torrent is already present"), tr("Torrent '%1' is already in the transfer list. Trackers have been merged.").arg(torrent->name()), QMessageBox::Ok);
+                QStringList message;
+
+                int newTrackersLen = torrent->addTrackers(m_torrentInfo.trackers());
+                int newUrlSeedsLen = torrent->addUrlSeeds(m_torrentInfo.urlSeeds());
+
+                if (newTrackersLen > 0)
+                    message << tr("%1 new tracker(s)").arg(newTrackersLen);
+
+                if (newUrlSeedsLen > 0)
+                    message << tr("%1 new url seed(s)").arg(newUrlSeedsLen);
+
+                RaisedMessageBox::information(this, tr("Torrent is already present"), tr("Torrent '%1' is already in the transfer list").arg(torrent->name()) + (message.count() > 0 ? (tr(", %1 have been merged.").arg(message.join(tr(" and ")))): "."), QMessageBox::Ok);
```

Comment:
I think `tr(" and ")` needs context for translators and maybe `tr(", %1 have been merged.")`as well.
Also could you break such long lines to 2 or more if needed? At `+ (...`
Other than that there should be a space before `:`

Reply:
I think it would be more convenient (both in terms of code and translation) to divide it into several sentences:
1. Torrent '%1' is already in the transfer list.
2. %1 new tracker(s) have been added to the torrent.
3. %1 new URL seed(s) have been added to the torrent.

It looks a bit redundant, but the game is worth the candle, IMO.
In practice, it is a bad idea to divide a translated sentence into several parts without using a format string. It can't be translated correctly into all languages. Don't forget that different languages may use different order of sentence members, so having something like `tr(", 
and")` is a clear mistake.